src/Raven.Server/Documents/Queries/QueryMetadata.cs

```diff
@@ -567,8 +582,49 @@ void AddInclude(QueryExpression include, string path, ref List<string> listToAdd
                 CompareExchangeValueIncludes = compareExchangeValueIncludes?.ToArray();
         }
 
-        private static ExplanationField CreateExplanationField(MethodExpression expression)
+        private void AddToRevisionsInclude(RevisionIncludeField revisionIncludes, MethodExpression expression, BlittableJsonReaderObject parameters)
         {
+            foreach (var queryExpression in expression.Arguments)
+            {
+                switch (queryExpression)
+                {
+                    case FieldExpression fe:
+                    {
+                        if (Query.From.Alias?.Value.Equals(fe.Compound[0].Value) == false)
```

Comment:
Talk with @aviv86 how to handle this. I think we do have a dedicated set of methods for extraction of those.